src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -735,6 +736,22 @@ static KinesisClientLease newKCLLease(Shard shard) {
         return newLease;
     }
 
+    static KinesisClientLease newKCLLeaseForChildShard(ChildShard childShard) throws InvalidStateException {
+        KinesisClientLease newLease = new KinesisClientLease();
+        newLease.setLeaseKey(childShard.getShardId());
+        List<String> parentShardIds = new ArrayList<String>(2);
+        if (!CollectionUtils.isNullOrEmpty(childShard.getParentShards())) {
+            parentShardIds.addAll(childShard.getParentShards());
+        } else {
+            throw new InvalidStateException("Unable to populate new lease for child shard " + childShard.getShardId()
+            + " because parent shards cannot be found.");
+        }
+        newLease.setParentShardIds(parentShardIds);
+        newLease.setOwnerSwitchesSinceCheckpoint(0L);
+        newLease.setCheckpoint(ExtendedSequenceNumber.TRIM_HORIZON);
```

Comment:
Looks like this line is the only substantial difference between this method and `newKCLLease`. Could we reuse code between the two and only set the TRIM_HORIZON checkpoint if the input is a childShard? Or is there a reason for separating the two methods?

Reply:
The main difference of newKCLLeaseForChildShard and newKCLLease is the parameter type. That's why we have a separate method for creating leases for child shards.